src/debug/debug_impl.ts

```diff
@@ -146,6 +146,20 @@ export class DartDebugSession extends DebugSession {
 							let thread = this.threadManager.getThreadInfoFromRef(isolateRef);
 							thread.receivedPauseStart();
 						}
+
+						// Helpers to categories libraries as SDK/ExternalLibrary/not.
+						let localPackageName = this.packageMap.getLocalPackageName();
+						let isSdkLibrary = (l: VMLibraryRef) => l.name.startsWith("dart.") || l.name == "builtin";
+						let isExternalLibrary = (l: VMLibraryRef) => !isSdkLibrary(l) && l.uri.startsWith("package:") && !l.uri.startsWith(`package:${localPackageName}/`);
```

Comment:
I'll have to check. When I first did it, without this some things would appear as both SDK and External Libraries, and that meant they were altered by either option rather than just the corresponding one. It was a few weeks ago though (before I had a good isExternalLibrary implementation) so might be irrelevant now.